pkg/query-service/app/http_handler.go

```diff
@@ -5650,3 +5659,535 @@ func (aH *APIHandler) getDomainInfo(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	response := traceFunnels.NewFunnelResponse{
+		ID:        funnel.ID,
+		Name:      funnel.Name,
+		CreatedAt: funnel.CreatedAt / 1000000,
+		CreatedBy: funnel.CreatedBy,
+		OrgID:     orgID,
+	}
+	json.NewEncoder(w).Encode(response)
+}
+
+// handleUpdateFunnelStep adds or updates steps for an existing funnel
+// Steps are identified by their step_order, which must be unique within a funnel
+func (aH *APIHandler) handleUpdateFunnelStep(w http.ResponseWriter, r *http.Request) {
+	var req traceFunnels.FunnelStepRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	claims, ok := authtypes.ClaimsFromContext(r.Context())
+	if !ok {
+		http.Error(w, "unauthenticated", http.StatusUnauthorized)
+		return
+	}
+	userID := claims.UserID
+
+	if err := traceFunnels.ValidateTimestamp(req.Timestamp, "updated_timestamp"); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	//aH.Signoz.SQLStore.SQLxDB()
+	dbClient, _ := traceFunnels.NewSQLClient(aH.Signoz.SQLStore)
+
+	funnel, err := dbClient.GetFunnelFromDB(req.FunnelID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("funnel not found: %v", err), http.StatusNotFound)
+		return
+	}
+
+	// Process each step in the request
+	for i := range req.Steps {
+		if req.Steps[i].StepOrder < 1 {
+			req.Steps[i].StepOrder = int64(i + 1) // Default to sequential ordering if not specified
+		}
+	}
+
+	if err := traceFunnels.ValidateFunnelSteps(req.Steps); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Normalize step orders
+	req.Steps = traceFunnels.NormalizeFunnelSteps(req.Steps)
+
+	// Update the funnel with new steps
+	funnel.Steps = req.Steps
+	funnel.UpdatedAt = req.Timestamp * 1000000
+	funnel.UpdatedBy = userID
+
+	// Update funnel in database
+	err = aH.TraceFunnels.UpdateFunnel(funnel, userID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to update funnel in database: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	response := traceFunnels.FunnelStepResponse{
+		FunnelID:  funnel.ID,
+		Name:      funnel.Name,
+		Steps:     funnel.Steps,
+		Timestamp: req.Timestamp,
+		CreatedAt: funnel.CreatedAt / 1000000,
+		CreatedBy: funnel.CreatedBy,
+		OrgID:     funnel.OrgID,
+		UpdatedBy: userID,
+	}
+
+	json.NewEncoder(w).Encode(response)
+}
+
+func (aH *APIHandler) handleListFunnels(w http.ResponseWriter, r *http.Request) {
+	claims, ok := authtypes.ClaimsFromContext(r.Context())
+	if !ok {
+		http.Error(w, "unauthenticated", http.StatusUnauthorized)
+		return
+	}
+	orgID := claims.OrgID
+
+	var dbFunnels []*traceFunnels.Funnel
+	var err error
+
+	dbClient, _ := traceFunnels.NewSQLClient(aH.Signoz.SQLStore)
+
+	if orgID != "" {
+		dbFunnels, err = dbClient.ListFunnelsFromDB(orgID)
+	} else {
+		dbFunnels, err = dbClient.ListAllFunnelsFromDB()
```

Comment:
use default org Id, no need to create a different function.

Reply:
done